features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/attachments/preview/AttachmentsPreviewPresenter.kt

```diff
@@ -93,13 +108,17 @@ class AttachmentsPreviewPresenter @AssistedInject constructor(
             mimeType = mediaAttachment.localMedia.info.mimeType,
             compressIfPossible = mediaAttachment.compressIfPossible,
             progressCallback = progressCallback
-        ).fold(
-            onSuccess = {
-                sendActionState.value = SendActionState.Done
-            },
-            onFailure = {
-                sendActionState.value = SendActionState.Failure(it)
+        ).getOrThrow()
```

Comment:
I am not sure. Is this `.getOrThrow()` necessary?

Reply:
We need it to catch the `CancellationException` that won't come inside the `Result`. I tried several ways to get it wrapped inside this Result, but none of them worked.